app/src/main/java/ch/hikemate/app/model/authentication/FirebaseAuthRepository.kt

```diff
@@ -127,4 +134,91 @@ class FirebaseAuthRepository : AuthRepository {
     Firebase.auth.signOut()
     onSuccess()
   }
+
+  override fun deleteAccount(
+      password: String,
+      activity: Activity,
+      onSuccess: () -> Unit,
+      onErrorAction: (Exception) -> Unit
+  ) {
+    val user = FirebaseAuth.getInstance().currentUser
+    if (user != null) {
+      val email = user.email ?: throw Exception("User email is null")
+      Log.d("DeleteAccount", "User email: '$email', password: '$password'")
+
+      // Re-authenticate the user
+      reauthenticate(
+          user,
+          password,
+          activity,
+          {
+            Log.d("DeleteAccount", "User re-authenticated")
+
+            // Delete the user profile and saved hikes
+            FirebaseFirestore.getInstance()
+                .collection(ProfileRepositoryFirestore.PROFILES_COLLECTION)
+                .document(user.uid)
+                .delete()
+            FirebaseFirestore.getInstance()
+                .collection(SavedHikesRepositoryFirestore.SAVED_HIKES_COLLECTION)
+                .document(user.uid)
+                .delete()
```

Comment:
Question: Do you have an easy way to check for correctness of the data deletion? I'm pretty sure Firebase should return a boolean somewhere to indicate that deletion has correctly happened

Reply:
Part of my concern is about the deletion of the Firebase user (and thus permission) before the deletion is finished, leading to permissions errors

Reply:
I fixed this. Note that there is a sonar warning about `Task<Void>` for which it suggest to change it to `Task<Unit>`. This doesn't work as the tasks types for firebase operations are `Task<Void>`. I suggest that we remove this check in sonar as this is not that bad of a practice.